transmission_interface/test/simple_transmission_loader_test.cpp

```diff
@@ -0,0 +1,382 @@
+      <joint name="joint1">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+      <transmission name="transmission1">
+        <plugin>transmission_interface/SimpleTransmission</plugin>
+        <joint name="joint1" role="joint1">
+          <!-- Unspecified element -->
+
+        </joint>
+      </transmission>
+
+      <joint name="joint2">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+      <transmission name="transmission2">
+        <plugin>transmission_interface/SimpleTransmission</plugin>
+        <joint name="joint2" role="joint2">
+          <mechanical_reduction>50</mechanical_reduction> <!-- Not a number -->
+        </joint>
+      </transmission>
+
+      <joint name="joint3">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+      <transmission name="transmission3">
+        <plugin>transmission_interface/SimpleTransmission</plugin>
+        <joint name="joint3" role="joint3">
+          <offset> 0</offset> <!-- Not a number -->
+          <mechanical_reduction>50</mechanical_reduction>
+        </joint>
+      </transmission>
+
+      <joint name="joint4">
+        <command_interface name="velocity">
+          <param name="min">-1</param>
+          <param name="max">1</param>
+        </command_interface>
+        <state_interface name="velocity"/>
+      </joint>
+      <transmission name="transmission4">
+        <plugin>transmission_interface/SimpleTransmission</plugin>
+        <joint name="joint4" role="joint4">
+          <mechanical_reduction>60</mechanical_reduction>           <!-- Invalid value -->
+        </joint>
+      </transmission>
+  </ros2_control>
+  </robot>
+)";
+  // Parse transmission info
+  std::vector<HardwareInfo> infos = parse_control_resources_from_urdf(urdf_to_test);
+  ASSERT_EQ(4lu, infos[0].transmissions.size());
+
+  // Transmission loader
+
+  std::vector<TransmissionPluginLoader> loader(4);
+  std::vector<std::shared_ptr<TransmissionLoader>> transmission_loader(4);
+
+  for (int i = 0; i < 4; ++i)
+  {
+    transmission_loader[i] = loader[i].create(infos[0].transmissions[i].type);
+    ASSERT_TRUE(nullptr != transmission_loader[i]);
+  }
+
+  for (int i = 0; i < 4; ++i)
+  {
+    std::shared_ptr<Transmission> transmission = nullptr;
+    transmission = transmission_loader[i]->load(infos[0].transmissions[i]);
+    ASSERT_TRUE(nullptr == transmission);
+  }
+}
+
+*/
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
```

Comment:
Is this really needed? We don't have this in the test files usually. At least the one using `gmock` instead of `gtest` library.

```suggestion
```

Reply:
I can fix this in #642 . The trick is to link your tests binary to a library called "gmock_main".

Reply:
fixed over there